project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -1413,18 +1442,10 @@ record ExperimentDeletionRequest(String projectId, String experimentId, String r
                                    ExperimentDeletionRequestBody body) {
 
     public ExperimentDeletionRequest {
-      if (projectId == null) {
-        throw new IllegalArgumentException("Project ID cannot be null");
-      }
-      if (projectId.isBlank()) {
-        throw new IllegalArgumentException("Project ID cannot be blank");
-      }
-      if (experimentId == null || experimentId.isBlank()) {
-        throw new IllegalArgumentException("Experiment ID cannot be empty");
-      }
-      if (requestId == null || requestId.isBlank()) {
-        requestId = UUID.randomUUID().toString();
-      }
+      requireNonNull(projectId);
+      requireNonNull(experimentId);
+      requestId =
+          isNull(requestId) || requestId.isBlank() ? UUID.randomUUID().toString() : requestId;
```

Comment:
i think this is a open door for nasty side effects. The client either provides the ID or must explicitly call the constructor method for creating an ID automatically.